e2e/machine_migration_capi_authoritative_test.go

```diff
@@ -161,11 +161,84 @@ var _ = Describe("[sig-cluster-lifecycle][OCPFeatureGate:MachineAPIMigration] Ma
 				})
 				It("should verify the AWS machine is deleted", func() {
 					verifyResourceRemoved(&awsv1.AWSMachine{
-						TypeMeta: metav1.TypeMeta{Kind: "AWSMachine", APIVersion: awsv1.GroupVersion.String()},
+						TypeMeta:   metav1.TypeMeta{Kind: "AWSMachine", APIVersion: awsv1.GroupVersion.String()},
 						ObjectMeta: metav1.ObjectMeta{Name: mapiMachineAuthCAPINameDeletion, Namespace: capiframework.CAPINamespace},
 					})
 				})
 			})
 		})
 	})
+
+	var _ = Describe("Machine Migration Round Trip Tests", Ordered, func() {
+		var capiMapiCapiRoundTripName = "machine-capi-mapi-capi-roundtrip"
+		var newMapiMachine *mapiv1beta1.Machine
+		var newCapiMachine *clusterv1.Machine
+
+		Context("CAPI -> MAPI -> CAPI round trip", func() {
+			BeforeAll(func() {
+				DeferCleanup(func() {
+					By("Cleaning up machine resources")
+					cleanupMachineResources(
+						ctx,
+						cl,
+						[]*clusterv1.Machine{newCapiMachine},
+						[]*mapiv1beta1.Machine{newMapiMachine},
+					)
+				})
+			})
+
+			It("should create MAPI Machine with spec.authoritativeAPI: ClusterAPI", func() {
+				newMapiMachine = createMAPIMachineWithAuthority(ctx, cl, capiMapiCapiRoundTripName, mapiv1beta1.MachineAuthorityClusterAPI)
+				newCapiMachine = capiframework.GetMachine(cl, capiMapiCapiRoundTripName, capiframework.CAPINamespace)
+				verifyMachineRunning(cl, newCapiMachine)
+			})
+
+			It("should wait for migration to complete", func() {
+				verifyMachineAuthoritative(newMapiMachine, mapiv1beta1.MachineAuthorityClusterAPI)
+				verifyMAPIMachineSynchronizedCondition(newMapiMachine, mapiv1beta1.MachineAuthorityClusterAPI)
+				verifyMachineSynchronizedGeneration(cl, newMapiMachine, mapiv1beta1.MachineAuthorityClusterAPI)
+				verifyMachinePausedCondition(newMapiMachine, mapiv1beta1.MachineAuthorityClusterAPI)
+				verifyMachinePausedCondition(newCapiMachine, mapiv1beta1.MachineAuthorityClusterAPI)
+			})
+
+			It("should update spec.authoritativeAPI: MachineAPI", func() {
```

Comment:
The controller doesn't change the authApi, we do it's not a controller behaviour - so IMO we want to capture this in the code / logs we output. 

I think if we rewrite out It() to describe the behaviour we want, and then trigger it with a code coming after a By() describing it, we're good :)